fluidefi/fluidefi-caspernet-aggregator-tools
# Asset Monitor Service

**Summary of change:**

- Added a new service, asset monitor service, which updates the all_pairs & token tables once a new PairCreated event is added to the datastore. 
- Added a new micro-service, PairCreatedEventsFinder, which finds all PairCreatedEvents for the given block height.
- Added a new micro-service, _PairMintEventFinder, which finds all PairMintEvents for the given address.
- created parser and saver class for erc20token, and uniswapV2Pair.
- created fetcher classes for token name, symbol, and decimals.

**List of tasks:** 

**_PairCreatedEventsFinder:_**


**_AllPairsFinder:_**

**_PairMintEventFinder:_**

**_UniswapV2PairSaver:_**

**_TokenNameFetcher:_**
- [x] create integration test. 

**_TokenSymbolFetcher:_**
- [x] create integration test. 

**_TokenDecimalsFinder:_**

**_ERC20TokenParser:_**

**_ERC20TokenSaver:_**

**_ERC20TokenFinder:_**

**_ERC20TokenResolver:_**

**_ERC20TokenUpdater:_**

**_AssetMonitorConsumer:_**


**Steps for testing**
- Make sure your local git is up to date
```bash
git pull
```

- Checkout the branch `feature/asset-monitor-service`
```bash
git checkout feature/asset-monitor-service
```

- Run the test across the test-net
```bash
npm run test
```

- Confirm tests have passed.
- Run example.
